gitlab/v3/objects.py

```diff
@@ -1675,6 +1675,7 @@ class ProjectService(GitlabObject):
 
                  # Optional fields
                  'username', 'password', 'jira_issue_transition_id')),
+        'mattermost': (('webhook',), ('username', 'channel')),
```

Comment:
I can't find references to `username` and `channel` in the gitlab documentation and code. Is this really supported?

Reply:
Looking at [services API](https://docs.gitlab.com/ee/api/services.html#mattermost-notifications), it looks like `username` and `channel` are optional fields.

Reply:
You're right. I guess I missed it.